cliquet_docs/reference/permission.rst

```diff
@@ -1,102 +1,182 @@
-| GET / HEAD | ``read``                                       |
-+------------+------------------------------------------------+
-| PUT        | ``create`` if record doesn't exist, ``write``  |
-|            | otherwise                                      |
-+------------+------------------------------------------------+
-| PATCH      | ``write``                                      |
-+------------+------------------------------------------------+
-| DELETE     | ``write``                                      |
-+------------+------------------------------------------------+
+When a record is created or modified, the current user is added to
+list of principals for the ``write`` permission. That means that a user is
+always able to replace or delete the records she created.
 
-Route factories are `best described in the pyramid documentation
-<http://docs.pylonsproject.org/projects/pyramid/en/latest/narr/urldispatch.html#route-factories>`_
+.. note::
 
-.. autoclass:: cliquet.authorization.RouteFactory
-  :members:
+    When defining permissions, there are two specific principals:
 
-Authorization policy
-====================
+    * ``system.Authenticated``: any authenticated user
+    * ``system.Everyone``: any user
 
-Upon access, the authorization policy is asked if any of the current list of
-:term:`principals` has access to the current resource. By default, the authorization
-policy *Cliquet* checks in the :term:`permission` backend for the current :term:`object`.
 
-It is possible to extend this behavior, for instance if there is an inheritance
-tree between the defined resources (some :term:`ACEs` should give access to its child
-:term:`objects`).
+Dynamic permissions
+-------------------
 
-In case the application should define its own inheritance tree, it should also
-define its own authorization policy.
+In the above section, the list of allowed principals for actions on the collection
+(especially ``create``) is specified via settings.
 
-To do so, subclass the default ``AuthorizationPolicy`` and add a specific
-``get_bound_permission`` method.
+One way of achieving dynamic permissions is to manipulate the permission backend
+manually.
+
+.. code-block:: python
+
+    def my_view(request):
+        permission = request.registry.permission
+
+        current_user_id = request.prefixed_userid
+        perm_object_id = '/{collection}'
+        permission.add_principal_to_ace(perm_object_id,
+                                        'create'
+                                        current_user_id)
+
+Alternatively, since :term:`principals` can be anything, it is also possible to use them to
+define groups:
+
+.. code-block:: python
+
+    def my_view(request):
+        permission = request.registry.permission
+
+        current_user_id = request.prefixed_userid
+        permission.add_user_principal(current_user_id, 'group:admins')
+
+
+And then refer as ``group:admins`` in the list of allowed principals.
```

Comment:
A use case could be workers that talk to cliquet to do async stuffs. 
Then could you explain how you set the ids of your worker and how do you set them to a `'group:workers'`